### PR TITLE
[4.x] Validation deprecations

### PR DESCRIPTION
**Pull request by burzum**
Refs #11934

### CODE REVIEW

File: src/Form/Form.php [hunk 1]
```diff
@@ -211,7 +166,7 @@ protected function _buildValidator(Validator $validator)
      */
     public function buildValidator(Event $event, Validator $validator, $name)
     {
-        $this->_buildValidator($validator);
+        $this->setValidator($name, $validator);
```
Comment (ADmad) [inline src/Form/Form.php]: `setValidator()` is already called by `getValidator()`. The call to `_buildValidator()` was only for BC, so now this method can just be empty.

Comment (ADmad) [reply src/Form/Form.php]: Actually this method as well as `implementedEvents()` can be simply removed. For normal use case one you just use `validationDefault()` method to configure validations as done in tables.

File: src/Form/Form.php [hunk 2]
```diff
@@ -224,7 +179,7 @@ public function validate(array $data)
     {
         $validator = $this->getValidator();
         if (!$validator->count()) {
```
Comment (ADmad) [inline src/Form/Form.php]: The whole `if` block needs to be removed.

**Review by ADmad — APPROVED** [review]

**Comment by burzum** [comment]
@ADmad @markstory can I merge it?

**Pull request merged**